rb/lib/selenium/webdriver/common/script.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module Selenium
+  module WebDriver
+    class Script
+      def initialize(bridge)
+        @bidi = bridge.bidi
+        @log_entry_subscribed = false
+      end
+
+      # @return [int] ID of the handler
+      def add_console_message_handler
+        subscribe_log_entry unless @log_entry_subscribed
+        @bidi.add_callback('log.entryAdded') do |params|
+          yield(params) if params['type'] == 'console'
+        end
+      end
+
+      # @return [int] ID of the handler
+      def add_javascript_error_handler
+        subscribe_log_entry unless @log_entry_subscribed
+        @bidi.add_callback('log.entryAdded') do |params|
+          yield(params) if params['type'] == 'javascript'
```

Comment:
For Java, we need to map the returned JSON result to a type so the end-user does not have to figure out how to use it or figure out the structure of the JSON. The IDE will provide suggestions with all the getter methods. I don't know if that applies to Ruby.

Reply:
Ideally we would generate the types for returned objects from BiDi spec, but we're far from there.

Reply:
I'm going to do a quick implementation with Struct and see how it looks.

Reply:
Implemented a Struct. I think this is the right compromise between wordiness and flexibility; if we create new objects for these later, the user is unlikely to need to change  anything the because the accessors should remain the same.